applications/nrf5340_audio/src/audio/streamctrl_unicast_server.c

```diff
@@ -0,0 +1,722 @@
+				break;
+			}
+
+			ret = audio_datapath_pres_delay_us_set(pres_delay_us);
+			if (ret) {
+				LOG_ERR("Failed to set presentation delay to %d", pres_delay_us);
+				break;
+			}
+
+			LOG_INF("Presentation delay %d us is set by initiator", pres_delay_us);
+
+			break;
+
+		case LE_AUDIO_EVT_NO_VALID_CFG:
+			LOG_WRN("No valid configurations found, will disconnect");
+
+			ret = bt_mgmt_conn_disconnect(msg.conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+			if (ret) {
+				LOG_ERR("Failed to disconnect: %d", ret);
+			}
+
+			break;
+
+		default:
+			LOG_WRN("Unexpected/unhandled le_audio event: %d", event);
+
+			break;
+		}
+
+		STACK_USAGE_PRINT("le_audio_msg_thread", &le_audio_msg_sub_thread_data);
+	}
+}
+
+/* Handle content control events */
+static void content_control_msg_sub_thread(void)
+{
+	int ret;
+
+	const struct zbus_channel *chan;
+
+	while (1) {
+		ret = zbus_sub_wait(&content_control_evt_sub, &chan, K_FOREVER);
+		ERR_CHK(ret);
+
+		struct content_control_msg msg;
+
+		ret = zbus_chan_read(chan, &msg, ZBUS_READ_TIMEOUT_MS);
+		ERR_CHK(ret);
+
+		uint8_t event = msg.event;
+
+		LOG_DBG("Received event = %d, current state = %d", event, strm_state);
+
+		switch (event) {
+		case MEDIA_PLAY:
+			break;
+
+		case MEDIA_STOP:
+			break;
+
+		default:
+			LOG_WRN("Unexpected/unhandled content control event: %d", event);
+			break;
+		}
+
+		STACK_USAGE_PRINT("content_control_msg_thread",
+				  &content_control_msg_sub_thread_data);
+	}
+}
+
+/**
+ * @brief	Zbus listener to receive events from bt_mgmt
+ *
+ * @note	Will in most cases be called from BT_RX context,
+ *		so there should not be too much processing done here
+ */
+static void bt_mgmt_evt_handler(const struct zbus_channel *chan)
+{
+	int ret;
+	const struct bt_mgmt_msg *msg;
+
+	msg = zbus_chan_const_msg(chan);
+	uint8_t event = msg->event;
+
+	switch (event) {
+	case BT_MGMT_CONNECTED:
+		LOG_INF("Connected");
+
+		break;
+
+	case BT_MGMT_DISCONNECTED:
+		LOG_INF("Disconnected");
+
+		/*TODO: le_audio_conn_disconnected(msg->conn); */
+
+		ret = bt_content_ctrl_conn_disconnected(msg->conn);
+		if (ret) {
+			LOG_ERR("bt_content_ctrl_disconnected failed with %d", ret);
+		}
+
+		break;
+
+	case BT_MGMT_SECURITY_CHANGED:
+		LOG_INF("Security changed");
+
+		ret = bt_rend_discover(msg->conn);
+		if (ret) {
+			LOG_WRN("Failed to discover rendering services");
+		}
+
+		ret = bt_content_ctrl_discover(msg->conn);
+		if (ret == -EALREADY) {
+			LOG_DBG("Discovery in progress or already done");
+		} else if (ret) {
+			LOG_ERR("Failed to start discovery of content control: %d", ret);
+		}
+
+		break;
+
+	case BT_MGMT_PA_SYNC_OBJECT_READY:
+		LOG_INF("PA sync object ready");
+
+		break;
+
+	default:
+		LOG_WRN("Unexpected/unhandled bt_mgmt event: %d", event);
+
+		break;
+	}
+}
+
+ZBUS_LISTENER_DEFINE(bt_mgmt_evt_sub, bt_mgmt_evt_handler);
+
+static int zbus_link(void)
+{
+	int ret;
+
+	if (IS_ENABLED(CONFIG_ZBUS) && (CONFIG_ZBUS_RUNTIME_OBSERVERS_POOL_SIZE > 0)) {
```

Comment:
It is better to have this check in the beginning:

if (!IS_ENABLED(CONFIG_ZBUS) || (CONFIG_ZBUS_RUNTIME_OBSERVERS_POOL_SIZE <= 0)) {
		return -ENOTSUP;
	}

Reply:
Discuss

Reply:
I will change this.